code/controllers/subsystem/security_level.dm

```diff
@@ -56,6 +56,13 @@ SUBSYSTEM_DEF(security_level)
 		SSshuttle.emergency.modTimer(old_shuttle_call_time_mod)
 		SSshuttle.emergency.modTimer(selected_level.shuttle_call_time_mod)
 
+//SKYRAT EDIT ADDITION START- In the event of an orange alert, change the airlock permissions
+	if(current_security_level.name == "orange")
+		enable_engineering_access()
+	else
+		revoke_engineering_access()
+//SKYRAT EDIT ADDITION END
+
 	SEND_SIGNAL(src, COMSIG_SECURITY_LEVEL_CHANGED, selected_level.number_level)
```

Comment:
please make engineering cards register to the security level change signal instead and update access based on that